control-plane/agents/src/bin/core/controller/scheduling/volume_group.rs

```diff
@@ -58,3 +58,28 @@ pub(crate) async fn get_pool_vg_replica_count(
     }
     pool_vg_replica_count
 }
+
+/// Get the map of node to the number of the volume group nexuses on the node.
+pub(crate) async fn get_node_vg_nexus_count(
+    volume_group_spec: &VolumeGroupSpec,
+    registry: &Registry,
+) -> HashMap<NodeId, u64> {
+    let mut node_vg_nexus_count = HashMap::new();
+    let vg_vols: Vec<String> = volume_group_spec
+        .volumes()
+        .iter()
+        .map(|vol_id| vol_id.to_string())
+        .collect();
+    let nodes = registry.specs().nodes();
+    for node in nodes {
+        if let Ok(nexuses) = registry.node_nexuses(node.id()).await {
```

Comment:
should use nexus specs here, rather than registry.
I think we already have 1 fn to get nexuses from volume identifier.

Reply:
But that might not be accurate right, for cases if there is no nexus for nexus spec? Then the distribution can get affected?

Reply:
It actually the other way around, spec is the proper way of finding out how many nexuses are meant to exist in a node. Let's a say a node restarts. temporarily it will say 0 nexuses!

Reply:
I see, okay. I will change it to use the spec.

Reply:
Done.